packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
+      header: style == DiagnosticsTreeStyle.singleLine ?
+        value?.join(', ') : value?.join('\n'),
+      showNull: showNull,
+      style: style,
+      hidden: value != null && value.isEmpty,
+    );
+  }
+
+  /// Diagnostics containing just a string `message` and not a a concrete name
+  /// or value.
+  factory DiagnosticsNode.message(String message, {
+    DiagnosticsTreeStyle style: DiagnosticsTreeStyle.singleLine,
+  }) {
+    return new _LeafDiagnosticsValue(
+      name: '',
+      header: message,
+      style: style,
+    );
+  }
+
+  /// Constructor that creates a [DiagnosticsNode] where properties and children
+  /// are computed lazily.
+  factory DiagnosticsNode.lazy({
+    String name,
+    Object object,
+    String header,
+    void fillProperties(List<DiagnosticsNode> properties),
+    List<DiagnosticsNode> getChildren(),
+    String emptyDescription,
+    DiagnosticsTreeStyle style=DiagnosticsTreeStyle.normal,
+  }) {
+    return new _LazyMembersDiagnosticsNode(
+      name: name,
+      object: object,
+      header: header,
+      fillProperties: fillProperties,
+      getChildren: getChildren,
+      style: style,
+      emptyDescription: emptyDescription,
+    );
+  }
+
+  /// Label describing the Diagnostics node.
+  ///
+  /// * If the DiagnosticsNode represents property, the [name] indicates the
+  ///   name of the property.
+  /// * If the DiagnosticsNode is a child of another DiagnosticsNode, the name
+  ///   describes the relationship between the parent and the child. For
+  ///   example: `child 5`.
+  final String name;
+
+  /// Header with a short text description of the node itself not including
+  /// children or properties.
+  final String header;
+
+  /// Whether to show if the value is null.
+  final bool showNull;
+
+  /// Whether the node is explicitly hidden.
+  final bool _hidden;
+
+  /// Whether to show a separator between [name] and [header].
+  ///
+  /// If false, name and header should be shown with no separation.
+  /// `:` is typically used as a separator when displaying as text.
+  final bool showSeparator;
+
+  /// Optional defaultValue the object has.
+  /// If [object] equals [defaultValue] and [defaultValue] is not null then
+  /// the node is [hidden].
+  final Object defaultValue;
+
+  /// Whether the diagnostics should be hidden when showing the default
+  /// view of a tree. Some displays of a tree could provide an affordance to
+  /// display hidden diagnostics.
+  bool get hidden {
+    if (_hidden)
+      return true;
+
+    if (defaultValue != null) {
+      try {
+        if (object == defaultValue)
+          return true;
+      } catch (e) {
+        // An object that throws an exception can't be null so the property
+        // should be displayed regardless.
+        return false;
+      }
+    }
+
+    if (showNull)
+      return false;
+
+    try {
+      return object == null;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /// Whether to show the name.
```

Comment:
Here and elsewhere: documentation that could have been written by someone who knew nothing about the code other than the names of the members being documented should be omitted or replaced with something more detailed. Omitting it is fine, it just adds it to the list of members we need to document. Having tautological documentation like this is not good though because it hides the lack of usefulness from our metrics.